js/bitfinex2.js

```diff
@@ -298,23 +298,30 @@ module.exports = class bitfinex2 extends bitfinex {
 
     async fetchOrderBook (symbol, limit = undefined, params = {}) {
         await this.loadMarkets ();
-        let orderbook = await this.publicGetBookSymbolPrecision (this.extend ({
+        const defaultPrecision = 'R0';
+        const request = {
             'symbol': this.marketId (symbol),
-            'precision': 'R0',
-        }, params));
-        let timestamp = this.milliseconds ();
-        let result = {
+            'precision': defaultPrecision,
+        };
+        if (limit) {
+            request['len'] = limit;
+        }
+        const fullRequest = this.extend (request, params);
+        const orderbook = await this.publicGetBookSymbolPrecision (fullRequest);
+        const timestamp = this.milliseconds ();
+        const result = {
             'bids': [],
             'asks': [],
             'timestamp': timestamp,
             'datetime': this.iso8601 (timestamp),
             'nonce': undefined,
         };
         for (let i = 0; i < orderbook.length; i++) {
-            let order = orderbook[i];
-            let price = order[1];
+            const order = orderbook[i];
+            const priceIndex = (fullRequest['precision'] === defaultPrecision) ? 1 : 0;
```

Comment:
I suggest do a little optimization and move this line out above this loop – it would be a bit faster if we set `priceIndex` just once, because it's `const`, rather than set it on every iteration of the loop conditionally. That will help slightly reduce the algorithmic complexity. If we keep it in mind throughout the lib – we can sometimes avoid suboptimal constructs and make the lib a bit faster in general.